pkg/reconciler/instances/connectivityproxy/command.go

```diff
@@ -30,10 +36,42 @@ type CommandActions struct {
 	copyFactory            []CopyFactory
 }
 
-func (a *CommandActions) Install(context *service.ActionContext) error {
-	err := a.install.Invoke(context.Context, context.ChartProvider, context.Task, context.KubeClient)
+func (a *CommandActions) InstallOnReleaseChange(context *service.ActionContext, app *appsv1.StatefulSet) error {
+	if app == nil || (app != nil && app.GetLabels() == nil) {
+		return a.installOnCondition(context, context.ChartProvider)
+	}
+
+	appName := app.Name
+	appRelease := app.GetLabels()[ReleaseLabelKey]
+
+	chartProviderWithFilter := context.ChartProvider.WithFilter(func(manifest string) (string, error) {
+		unstructs, err := internalKubernetes.ToUnstructured([]byte(manifest), true)
+		if err != nil {
+			return "", errors.Wrapf(err, "while casting manifest to kubernetes unstructured")
+		}
+
+		for _, unstruct := range unstructs {
+			if unstruct != nil &&
+				unstruct.GetName() == appName &&
+				unstruct.GetKind() == ConnectivityProxyKind {
+
+				if unstruct.GetLabels() == nil || unstruct.GetLabels()[ReleaseLabelKey] == "" {
+					return "", errors.Errorf("Component does not have any release labels")
+				} else if unstruct.GetLabels()[ReleaseLabelKey] != appRelease {
+					return manifest, nil
+				}
+			}
+		}
+		return "", nil
+	})
+
+	return a.installOnCondition(context, chartProviderWithFilter)
```

Comment:
You do not need `chartWithProviderFilter`, when calling `WithFilter` the receiver is a pointer, you are adding the filter directly to `context.ChartProvider`.
Therefore passing `context`  is enough.

The reason to return the provider in the "With" functions is the ability to do fluent programming (e.g. `x.WithA().WithB().WithC()`)